packages/backend-plugin-api/report.api.md

```diff
@@ -24,6 +25,80 @@ import { Readable } from 'stream';
 import { Request as Request_2 } from 'express';
 import { Response as Response_2 } from 'express';
 
+// @public
+export type AuditorEvent = [
+  message: string,
+  meta: {
+    actor: AuditorEventActorDetails;
+    eventName: string;
+    stage: string;
+    meta?: JsonObject;
+    request?: AuditorEventRequest;
+  } & AuditorEventStatus,
+];
+
+// @public (undocumented)
+export type AuditorEventActorDetails = {
+  actorId?: string;
+  ip?: string;
+  hostname?: string;
+  userAgent?: string;
+};
+
+// @public
+export type AuditorEventArgs<T extends JsonObject> = {
+  message: string;
+  eventName: string;
+  stage: string;
+  request?: Request_2;
+  actorId?: string;
+  meta?: T;
+} & AuditorEventStatus;
+
+// @public
+export type AuditorEventFailureStatus<E = ErrorLike> = {
+  status: 'failed';
+  errors: E[];
+};
+
+// @public (undocumented)
+export type AuditorEventRequest = {
+  url: string;
+  method: string;
+};
+
+// @public (undocumented)
+export type AuditorEventStatus =
+  | AuditorEventSuccessStatus
+  | AuditorEventFailureStatus
+  | AuditorEventUnknownStatus;
+
+// @public
+export type AuditorEventSuccessStatus = {
+  status: 'succeeded';
+};
+
+// @public
+export type AuditorEventUnknownStatus = {
+  status: 'unknown';
+};
+
+// @public
+export interface AuditorService {
+  // (undocumented)
+  child(meta: JsonObject): AuditorService;
+  // (undocumented)
+  debug<T extends JsonObject>(args: AuditorEventArgs<T>): Promise<void>;
```

Comment:
Something to feed back into the BEP as well: how come we need multiple levels for auditor service, I'd expect there to only be a single level for audit events?

Reply:
Multiple logging levels in an audit service—such as debug, info, warn, and error—are essential for providing flexibility and depth in tracking, managing, and analyzing audit events. While it may seem intuitive to use a single level for audit logs, having multiple levels serves several important purposes:

1. **Varying Degrees of Detail**: Different levels capture different amounts of information, depending on the situation. 
   - **Debug**: Offers granular details such as API calls, database queries, or internal parameters, useful for troubleshooting and diagnosing issues.
   - **Info**: Logs high-level, significant events such as successful logins or configuration changes, which are key for compliance and routine audits.
   - **Warn**: Highlights non-critical issues, such as blocked access attempts or slow performance, which may indicate potential risks.
   - **Error**: Records critical failures that affect system integrity, like failed transactions or security breaches, essential for incident response.

2. **Performance Optimization**: Logging everything at a single verbose level (e.g., debug) can overload the system and waste storage. By categorizing events, you can adjust the verbosity based on the environment, keeping logs leaner and more manageable.

3. **Troubleshooting Efficiency**: Having different levels allows developers or auditors to quickly filter logs to focus on relevant details. For example, during error investigations, debug logs can reveal in-depth information, while info logs provide a broader context.

4. **Compliance Flexibility**: Audit logs often need to meet regulatory requirements. With multiple levels, organizations can focus on what’s required for compliance (e.g., info logs) while also keeping more detailed records for internal purposes.

5. **Incident Response and Analysis**: Different levels help reconstruct timelines and identify root causes. For instance, error and warn logs are essential for understanding critical failures, while debug logs offer more detailed traces for thorough investigation.

6. **Efficient Log Retention**: Higher-level logs like info or warn are typically retained for longer periods, as they are essential for audit trails. Debug logs, being more detailed and voluminous, are often kept only for short-term troubleshooting.

Reply:
@Rugvip I can add this information to the BEP if you'd like.

Reply:
As discussed in the framework SIG, let's move this to single method instead